help-center-wp-admin-disconnected: Remove react

mreishus opened the pull request:
## Proposed Changes / Why are these changes being made?

* This removes the usage of react in `help-center-wp-admin-disconnected`.

In some cases, this script looks to be the only usage of react on the front-end, and it appears that react isn't even needed. The script looks to only add an event-listener to an existing dom element. This should be possible without react.

The relevant JP file is here: https://github.com/Automattic/jetpack/blob/e7157103862f9c8a026f411b6f41475e74b54b76/projects/packages/jetpack-mu-wpcom/src/features/help-center/class-help-center.php#L424-L427

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Run `yarn dev --sync` in apps/help-center.
- Sandbox widgets.wp.com
- While logged in, visit the frontpage of a simple site you own.
- Click the Help Center.
- It should open a new tab.
- The current tab should emit a tracks event `calypso_inlinehelp_show`.
- The new tab should have the Help Center open.
 
(Note that for it to be loaded, you have to view the front-end of a site while being logged in and having edit capabilities. )

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

mreishus committed:
help-center-wp-admin-disconnected: Remove react

alshakero commented:
@mreishus I added testing steps.

mreishus commented:
> @mreishus I added testing steps.
> 
> Edit: They're incorrect. I'll have to investigate tomorrow.

Thank you! I'll check it out tomorrow too.
Note that even with wp-admin and disconnected in the filename, I came across this when looking at the front-end of simple sites. I think https://github.com/Automattic/jetpack/pull/38941 expanded how this is used.

alshakero reviewed (approved):
I updated the testing steps and tested. Looks and works great. 

It still has a dependency on wp-polyfill for some reason though. See p1741780449118959-slack-C02T4NVL4JJ for a similar case. To me it's worth it to remove tracking if that's what it takes to get rid of the polyfill.

mreishus commented:
> I updated the testing steps and tested. Looks and works great.
> 
> It still has a dependency on wp-polyfill for some reason though. See p1741780449118959-slack-C02T4NVL4JJ for a similar case. To me it's worth it to remove tracking if that's what it takes to get rid of the polyfill.

Interesting! Thanks for your help. I found the polyfill was being added even if I changed the script to only "console.log('test');".  It looks like it's from this argument in the webpack:
https://github.com/Automattic/wp-calypso/blob/8ab299880163f92afa67904a143fad0f79db8468/apps/help-center/webpack.config.js#L65

Not sure what to do yet, but I'll proceed with this as it's an improvement.

mreishus commented:
177441-ghe-Automattic/wpcom
